src/analyses/goto_check.cpp

```diff
@@ -1113,7 +1113,8 @@ void goto_checkt::pointer_validity_check(
   const exprt &pointer=expr.pointer();
 
   auto size_of_expr_opt = size_of_expr(expr.type(), ns);
-  CHECK_RETURN(size_of_expr_opt.has_value());
+  if(!size_of_expr_opt.has_value())
+    return; // in the case of `void*`
```

Comment:
Shouldn't `size_of_expr()` also work for void pointers? In that case we should fix `size_of_expr()` is possible.

Reply:
But even if we implement size-of for `void` the result will be `0`, right? I'm not sure we should build the `address-check` in that case.

Reply:
`sizeof(void)` is undefined in C, some compilers set it to 1 just because there's too much code out there that assumes that `void* == char*`.

Reply:
Ah, I see. It's surprising that we get a dereference expression of a void pointer here.

Reply:
```
__CPROVER__start {
  char tmp;
  void *arg;
  if(NONDET(__CPROVER_bool))
    arg = NULL;

  else
  {
    arg = (void *)&tmp;
    *((void *)&tmp) = NONDET(void);
  }
  foo(arg);
}
```
This is what the entry-point looks like.

Reply:
Wouldn't `address_check` below with a zero-sized object still do some good?